js/screepsstats.js

```diff
@@ -54,212 +47,183 @@ ScreepsStats.prototype.addStat = function (key, value) {
 ScreepsStats.prototype.runBuiltinStats = function () {
 
   this.clean()
-  stats = {}
-  stats.time = new Date().toISOString()
-  stats.tick = Game.time
-
-  stats['cpu'] = {
-    'limit': Game.cpu.limit,
-    'tickLimit': Game.cpu.tickLimit,
-    'bucket': Game.cpu.bucket
-  }
-
-  stats['gcl'] = {
-    'level': Game.gcl.level,
-    'progress': Game.gcl.progress,
-    'progressTotal': Game.gcl.progressTotal
-  }
-
-  if(!stats['rooms']) {
-    stats['rooms'] = {
-      'subgroups': true
+  var stats = {
+    time: new Date().toISOString(),
+    tick: Game.time,
+    cpu: {
+      limit: Game.cpu.limit,
+      tickLimit: Game.cpu.tickLimit,
+      bucket: Game.cpu.bucket
+    },
+    gcl: {
+     level: Game.gcl.level,
+     progress: Game.gcl.progress,
+     progressTotal: Game.gcl.progressTotal
     }
   }
 
+  _.defaults(stats, {
+    rooms: {
+      subgroups: true
+    }
+  })
 
-  for(var roomName in Game.rooms) {
-    var room = Game.rooms[roomName]
+  _.forEach(Game.rooms,(room) => {
 
-    if(!stats[roomName]) {
-      stats['rooms'][roomName] = {}
+    if(!stats[room.name]) {
+      stats.rooms[room.name] = {}
     }
 
-    if(!!room.controller) {
-      var controller = room.controller
```

Comment:
I use _.isEmpty for all my undefined / null / "" / etc checks
keeps things simple